block/serialize.go

```diff
@@ -0,0 +1,301 @@
+package block
+
+import (
+	"github.com/vulpemventures/go-elements/internal/bufferutil"
+)
+
+func (b *Block) SerializeBlock() ([]byte, error) {
+	s, err := bufferutil.NewSerializer(nil)
+	if err != nil {
+		return nil, err
+	}
+
+	err = b.Header.SerializeHeader(s)
+	if err != nil {
+		return nil, err
+	}
+
+	err = b.TransactionsData.SerializeTransactions(s)
+	if err != nil {
+		return nil, err
+	}
+
+	return s.Bytes(), nil
+}
+
+func (t *Transactions) SerializeTransactions(
+	s *bufferutil.Serializer,
+) error {
+	err := s.WriteVarInt(uint64(len(t.Transactions)))
+	if err != nil {
+		return err
+	}
+	for _, v := range t.Transactions {
+		txBytes, err := v.Serialize()
+		if err != nil {
+			return err
+		}
+
+		err = s.WriteSlice(txBytes)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (h *Header) SerializeHeader(
+	s *bufferutil.Serializer,
+) error {
+	if h.ExtData.IsDyna {
+		err := s.WriteUint32(dynaVersion)
+		if err != nil {
+			return err
+		}
+	} else {
+		err := s.WriteUint32(proofVersion)
+		if err != nil {
+			return err
+		}
+	}
+
+	err := s.WriteSlice(h.PrevBlockHash)
+	if err != nil {
+		return err
+	}
+
+	err = s.WriteSlice(h.MerkleRoot)
+	if err != nil {
+		return err
+	}
+
+	err = s.WriteUint32(h.Timestamp)
+	if err != nil {
+		return err
+	}
+
+	err = s.WriteUint32(h.Height)
+	if err != nil {
+		return err
+	}
+
+	err = h.ExtData.serialize(s)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (e *ExtData) serialize(
```

Comment:
I think it would be better to make the ExtData methods public.

Reply:
@altafan  do u think there will be need to serialize something from ExtData separately from Header?

Reply:
I'm sorry, I was thinking ExtData was at the same level of Header and Transactions. It's fine as it is now. Resolving.